src/http_server/api/v1/metrics.c

```diff
@@ -211,23 +267,48 @@ void cb_metrics_prometheus(mk_request_t *request, void *data)
 
                 mk = sv.via.map.ptr[m].key;
                 mv = sv.via.map.ptr[m].val;
-
-                sds = flb_sds_cat(sds, "fluentbit_", 10);
-                sds = flb_sds_cat(sds, k.via.str.ptr, k.via.str.size);
-                sds = flb_sds_cat(sds, "_", 1);
-                sds = flb_sds_cat(sds, mk.via.str.ptr, mk.via.str.size);
-                sds = flb_sds_cat(sds, "_total{name=\"", 13);
-                sds = flb_sds_cat(sds, sk.via.str.ptr, sk.via.str.size);
-                sds = flb_sds_cat(sds, "\"} ", 3);
+                sds_metric = flb_sds_create_size(1024);
```

Comment:
note that sds_metric is never freed at the end of the function, so it's leaking memory